Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -160,31 +156,10 @@ struct CombinatorialKalmanFilterOptions {
   /// @param gctx The geometry context for this track finding/fitting
   /// @param mctx The magnetic context for this track finding/fitting
   /// @param cctx The calibration context for this track finding/fitting
-  /// @param accessor_ The source link accessor
-  /// @param extensions_ The extension struct
-  /// @param pOptions The plain propagator options
-  /// @param rSurface The reference surface for the eventual track fitting to be
-  /// expressed at
-  /// @param mScattering Whether to include multiple scattering
-  /// @param eLoss Whether to include energy loss
-  /// @param rSmoothing Whether to run smoothing to get fitted parameter
   CombinatorialKalmanFilterOptions(
       const GeometryContext& gctx, const MagneticFieldContext& mctx,
-      std::reference_wrapper<const CalibrationContext> cctx,
-      SourceLinkAccessor accessor_,
-      CombinatorialKalmanFilterExtensions<traj_t> extensions_,
-      const PropagatorPlainOptions& pOptions, const Surface* rSurface = nullptr,
-      bool mScattering = true, bool eLoss = true, bool rSmoothing = true)
-      : geoContext(gctx),
-        magFieldContext(mctx),
-        calibrationContext(cctx),
-        sourcelinkAccessor(std::move(accessor_)),
-        extensions(extensions_),
-        propagatorPlainOptions(pOptions),
-        smoothingTargetSurface(rSurface),
-        multipleScattering(mScattering),
-        energyLoss(eLoss),
-        smoothing(rSmoothing) {}
```

Comment:
There do all these go?

Reply:
to me the constructor seemed overloaded so I reduced it to a minimum as setting params after the constructor looks more explicit and readable to me. but that is only my preference

I guess extensions and source link accessor are mandatory so maybe they should remain. but the interactions flags and target surface and propagation options seem way to specialized to me

Reply:
I wondered about this when I saw the `extensions` and `propagatorPlainOptions` set in the `Examples` `TrackFindingAlgorithm`. There is the small complication/overhead of initialising those in the constructor and then overwriting them later.